Solutions/Cloud Service Threat Protection Essentials/Hunting Queries/AzureResourceAssignedPublicIP.yaml

```diff
@@ -69,5 +69,24 @@ query: |
   )
   | project StartTimeUtc, EndTimeUtc, FirstProcessedTimeUTC, LastProcessedTimeUtc, PublicIPs, NSG, NSG_Name, SrcIP, DestPort, SubNet, Name, VMs, MACAddresses, ActivityCount, Regions, AzureRegions, Subscription, Tags_s, SchemaVersion
   ) on $left.Resource == $right.NSG_Name
-  | extend timestamp = StartTimeUtc, AccountCustomEntity = Caller, IPCustomEntity = CallerIpAddress
+  | extend UserName = iff(Caller has '@', tostring(split(Caller, '@')[0]), '')
```

Comment:
Instead of has , contains fit best

Reply:
'has' will not match in case of 'publicIPAddressVersion', that's why 'contains' is used.